packages/flutter/lib/src/material/list_tile.dart

```diff
@@ -441,11 +456,13 @@ class ListTile extends StatelessWidget {
     return new InkWell(
       onTap: enabled ? onTap : null,
       onLongPress: enabled ? onLongPress : null,
-      child: new Container(
-        height: tileHeight,
-        padding: const EdgeInsets.symmetric(horizontal: 16.0),
-        child: new Row(children: children),
-      )
+      child: new ConstrainedBox(
+        constraints: new BoxConstraints(minHeight: tileHeight),
+        child: new Padding(
+          padding: const EdgeInsets.symmetric(horizontal: 16.0),
+          child: new IntrinsicHeight(child: new Row(children: children)),
```

Comment:
We really REALLY want to avoid using IntrinsicHeight in the framework itself if at all possible, it's massively expensive.

Reply:
(in the worst case it turns layout from linear to quadratic)

Why do you need an IntrinsicHeight here? Won't the Row just size itself to fit its children anyway?

Reply:
You would think so.  But in list_tile_test.dart, it sizes itself to [800x600 here](https://github.com/flutter/flutter/blob/4e48a737eb4e0e689e52678a50d5f15ee26c5b0f/packages/flutter/test/material/list_tile_test.dart#L104), rather than 800x56, without the IntrinsicHeight.

This is another case of possibly breaking customer code.  I think that's probably more justified here than in the button widget, since it's less likely to simply contain a single widget, but I was still loathe to do that.